src/components/ConversionRate/ConversionRateComponent.tsx

```diff
@@ -0,0 +1,319 @@
+
+  const handleKeyDown = (event: any) => {
+    if (event.key === 'Enter') {
+      setSearchConversionRate(event.currentTarget.value)
+      handleGetAllConversionRates()
+    }
+  }
+  const handleGetAllConversionRates = async () => {
+    try {
+      setIsLoading(true)
+      const res = await getAllConversionRates()
+
+      setConversionRates(res)
+    } catch (error) {
+      console.error(error)
+    } finally {
+      setIsLoading(false)
+    }
+  }
+
+  const handleAddNewConversionRate = async () => {
+    if (!fromCurrency) {
+      setFromCurrencyError('From currency cannot be empty')
+    }
+    if (!exchangeRate) {
+      setExchangeRateError('Exchange rate cannot be empty')
+    }
+    try {
+      await addNewCurrencyRate(fromCurrency, 'Credit', Number(exchangeRate))
+      Swal.fire({
+        title: 'Successfully added new conversion rate!',
+        icon: 'success',
+      })
+      setShouldUpdate((prev) => !prev)
+    } catch (error) {
+      console.error(error)
+    } finally {
+      setFromCurrencyError('')
+      setExchangeRateError('')
+    }
+  }
+
+  const handleShowSelectedConversionRate = async (conversionRateId: number) => {
+    const selectedConversionRate = conversionRates.find(
+      (cr: any) => cr.conversionRateId === conversionRateId,
+    )
+    setSelectedConversionRate(selectedConversionRate)
+    open()
+  }
+
+  const handleEditConversionRate = async () => {
+    try {
+      await editConversionRate(
+        selectedConversionRate!.conversionRateId,
+        Number(exchangeRate),
+      )
+      Swal.fire({
+        title: 'Successfully updated conversion rate!',
+        icon: 'success',
+      })
+      setShouldUpdate((prev) => !prev)
+    } catch (error) {
+      console.error(error)
+    } finally {
+      close()
+      setSelectedConversionRate(undefined)
+      setExchangeRate('')
+    }
+  }
+
+  const handleDeleteConversionRate = async (conversionRateId: number) => {
+    Swal.fire({
+      title: 'Are you sure to delete this conversion rate?',
+
+      icon: 'question',
+      showCancelButton: true,
+      confirmButtonColor: '#3085d6',
+      cancelButtonColor: '#d33',
+      confirmButtonText: 'Yes, delete!',
+    }).then(async (result) => {
+      if (result.isConfirmed) {
+        try {
+          await deleteConversionRate(conversionRateId)
+          Swal.fire({
+            title: 'Successfully deleted conversion rate!',
+            icon: 'success',
+          })
+          setShouldUpdate((prev) => !prev)
+        } catch (error) {
+          console.error(error)
+        }
+      }
+    })
+  }
+
+  useEffect(() => {
+    handleGetAllConversionRates()
+  }, [shouldUpdate])
+
+  const rows =
+    conversionRates.length > 0 ? (
+      conversionRates.map((cr: any) => (
+        <Table.Tr key={cr.conversionRateId} className="text-base h-16">
+          <Table.Td
+            onClick={() =>
+              handleShowSelectedConversionRate(cr.conversionRateId)
```

Comment:
We can use 'tr click' instead of 'td click

Reply:
it covers the whole tr and i couldn't click on other actions :))

Reply:
fixed